### ISSUE TITLE
GLPI 9.2.1 Wrong renames of pdf attachments

### ISSUE THREAD

**Issue by okvok** [glpi-project/glpi#3269]
Hello!
I've just faced with a strange behavior of GLPI 9.2.1. See screenshots below:
Sending email with attachments:
![image](https://user-images.githubusercontent.com/20252855/33818585-daae2394-de56-11e7-9113-d05f58af0f5e.png)

In registered ticket:
![image](https://user-images.githubusercontent.com/20252855/33818676-450c9b62-de57-11e7-8209-16475efb47ac.png)

A BUG ? All documents names were in Russian.

**Comment by okvok** [comment]
Sorry. It was a bug in my system. After reinstalling everything is OK.

**Comment by okvok** [comment]
Information about system installation and configuration
[code]
 
GLPI 9.2.1 ( => /var/www/glpi)
Installation mode: TARBALL

--------------------------------------------------
Server

Operating system: Linux bd-sd01.gpnbadra.local 3.10.0-693.11.1.el7.x86_64 #1 SMP Mon Dec 4 23:52:40 UTC 2017 x86_64
PHP 7.0.18 apache2handler (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apache2handler, apc, apcu, bz2, calendar,
	ctype, curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, imap, json, ldap, libxml, mbstring, mcrypt,
	mysqli, mysqlnd, openssl, pcre, pdo_mysql, pdo_sqlite, session, snmp, soap, sockets, sqlite3, standard, tokenizer, wddx, xml,
	xmlreader, xmlrpc, xmlwriter, xsl, zlib)
Setup: max_execution_time="30" memory_limit="128M" post_max_size="8M" safe_mode="" session.save_handler="files"
	upload_max_filesize="5M" 
Software: Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips PHP/7.0.18 ()
	Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.84 Safari/537.36
Server Software: MariaDB Server
	Server Version: 10.1.28-MariaDB
	Server SQL Mode: 
	Parameters: glpidb@GLPI_DB_prod/db_glpi_prod
	Host info: GLPI_DB_prod via TCP/IP
	
mysqli extension is installed
ctype extension is installed
fileinfo extension is installed
json extension is installed
mbstring extension is installed
zlib extension is installed
curl extension is installed
gd extension is installed
simplexml extension is installed
xml extension is installed
ldap extension is installed
imap extension is installed
Zend OPcache extension is installed
APCu extension is installed
xmlrpc extension is installed
OK/var/www/glpi/config : OK
OK/var/www/glpi/files : OK
OK/var/www/glpi/files/_dumps : OK
OK/var/www/glpi/files/_sessions : OK
OK/var/www/glpi/files/_cron : OK
OK/var/www/glpi/files/_graphs : OK
OK/var/www/glpi/files/_lock : OK
OK/var/www/glpi/files/_plugins : OK
OK/var/www/glpi/files/_tmp : OK
OK/var/www/glpi/files/_cache : OK
OK/var/www/glpi/files/_rss : OK
OK/var/www/glpi/files/_uploads : OK
OK/var/www/glpi/files/_pictures : OK
OK/var/www/glpi/files/_log : OK
OKSELinux mode is Enforcing
OKSELinux boolean configuration for httpd_can_network_connect --> on
OKSELinux boolean configuration for httpd_can_network_connect_db --> on
OKSELinux boolean configuration for httpd_can_sendmail --> on

-----------------------------------------------------------------
fields               Name: Additionnal fields             Version: 1.7.0      State: Enabled
	behaviors            Name: Behaviours                     Version: 2.0.0      State: Enabled
	browsernotification  Name: Browser Notification           Version: 1.1.9      State: Enabled
	dashboard            Name: Dashboard                      Version: 0.8.8      State: Enabled
	datainjection        Name: File injection                 Version: 2.4.2      State: Not activated
	fpsoftware           Name: FP Software                    Version: 1.2.0      State: Enabled
	fusioninventory      Name: FusionInventory                Version: 9.2+1.0    State: Enabled
	mailanalyzer         Name: Mail Analyzer                  Version: 1.3.2      State: Enabled
	genericobject        Name: Objects management             Version: 2.4.1      State: Enabled
	ticketcleaner        Name: Ticket Cleaner                 Version: 2.0.4      State: Not activated

**Comment by okvok** [comment]
And also I've applied some patches : 
------------------------------------------------------------------------------------------------------
https://github.com/glpi-project/glpi/issues/2869

The problem probably comes from here :

Modify

inc/html.class.php:5466
$fullpath = GLPI_DOC_DIR."/".$image['filepath'];
to
$fullpath = "/".$image['filepath'];

The inc/html.class.php:5466 fixes some of the issue but not all, inline mailed messages still have the wrong url, fix:

inc/html.class.php:5474

             if ($addLink) {
                $out .= '<a href="'.$CFG_GLPI['root_doc'].
                        '/front/document.send.php?docid='.$id.
                        '" target="_blank"><img alt="'.$image['tag'].
                        '" height="'.$height.'" width="'.$width.
                        '" src="'.$CFG_GLPI['root_doc'].
                   '/front/document.send.php?docid='.$id.'" /></a>';
             } else {
                $out .= '<img alt="'.$image['tag'].
                        '" height="'.$height.'" width="'.$width.
                        '" src="'.$CFG_GLPI['root_doc'].
                        '/front/document.send.php?docid='.$id.'" />';
to

             if ($addLink) {
                $out .= '<a href="/front/document.send.php?docid='.$id.
                        '" target="_blank"><img alt="'.$image['tag'].
                        '" height="'.$height.'" width="'.$width.
                        '" src="/front/document.send.php?docid='.$id.'" /></a>';
             } else {
                $out .= '<img alt="'.$image['tag'].
                        '" height="'.$height.'" width="'.$width.
                        '" src="/front/document.send.php?docid='.$id.'" />';
		}

**Comment by okvok** [comment]
I've changed some fields names in the interface through editing locales files en_GB.* with Poedit-1.7.7. software.
Here is changed fields list in Assets form:
Alternate username number
Supplier
Order number
Delivery form
Immobilization number

**Comment by AnaelMobilia** [comment]
Hi @okvok ,

It's hard to follow your ticket :)
Could you, please, attach to this case the pdf file which was renamed ?
This problem appears to another tickets, always with cyrilic alphabet (@aircraftru : you faces the same problem, no ?)

Stay in mind, that quick fix on your GLPI installation can have back effects with a future release of GLPI... ;-)
I understand the need of "quick fix", and I suggest you to contribute / backport to the project (glpi / transiflex) your updates.

Regards,
Anael

**Comment by okvok** [comment]
Hi,
I am on my vacation now )))
But you can randomly rename any pdf file. In my case it has got the bug in 100% )))
Unfortunately I couldn’t find the similar AircraftRu’s ticket. Could you prompt me his bug number please?

**Comment by johapihe** [comment]
it worked for me adding the following in this line, since I could not find the complete path

if ($addLink) {
            $out .= '<a href="**/glpi/**front/document.send.php?docid='.$id.
                    '" target="_blank"><img alt="'.$image['tag'].
                    '" height="'.$height.'" width="'.$width.
                    '" src="**/glpi/**front/document.send.php?docid='.$id.'" /></a>';
         } else {
            $out .= '<img alt="'.$image['tag'].
                    '" height="'.$height.'" width="'.$width.
                    '" src="**/glpi/**front/document.send.php?docid='.$id.'" />';
	}

**Comment by trasher** [comment]
Duplicates https://github.com/glpi-project/glpi/issues/3653
